_rules/invalid-form-field-value-36b590.md

```diff
@@ -0,0 +1,220 @@
+---
+id: 36b590
+name: Error message describes invalid form field value
+rule_type: atomic
+description: |
+  This rule checks that text descriptions are provided when the user completes a form field with invalid values or using an invalid format.
+accessibility_requirements:
+  wcag20:3.3.1: # Error Identification (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
```

Comment:
I've put several hours and a lot of thought into this. I don't think the current form works, for quite a few reasons. I'll outline a few of them as I go. But here's my counter-proposal:

```md
## Applicability

The rule applies to each HTML element that has one of the following [semantic roles][semantic role]: `checkbox`, `combobox`, `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` or `textbox`.

**Note**: The list of applicable [semantic roles][semantic role] is derived by taking all the [ARIA 1.1][] roles that:

- inherit from the [abstract][] `input` or `select` role, and
- do not have a [required context][] role that itself inherits from one of those roles.

## Expectation 1

For each test target with a [form field error indicator][], at least one of the [form field error indicators][] allows the identification of the related test target, through [text][], or [non-text content][], or through [presentation][].

## Expectation 2

For each test target with a [form field error indicator][], at least one of the [form field error indicators][] describes the cause of the error, or how to resolve it, in [text][] that is [visible][].

## Expectation 3

For each test target with a [form field error indicator][], at least one of the [form field error indicators][] describes the cause of the error, or how to resolve it, in [text][] that is [included in the accessibility tree][] or included in the [accessible name][] or [accessible description][] of the test target.
```